includes/class-amp-theme-support.php

```diff
@@ -1966,18 +1966,25 @@ public static function prepare_response( $response, $args = [] ) {
 			header( 'Content-Type: text/html; charset=utf-8' );
 		}
 
-		// @todo Both allow_dirty_styles and allow_dirty_scripts should eventually use AMP dev mode instead.
 		$args = array_merge(
 			[
 				'content_max_width'    => ! empty( $content_width ) ? $content_width : AMP_Post_Template::CONTENT_MAX_WIDTH, // Back-compat.
 				'use_document_element' => true,
-				'allow_dirty_styles'   => self::is_customize_preview_iframe(), // Dirty styles only needed when editing (e.g. for edit shortcuts).
-				'allow_dirty_scripts'  => is_customize_preview(), // Scripts are always needed to inject changeset UUID.
 				'user_can_validate'    => AMP_Validation_Manager::has_cap(),
 			],
 			$args
 		);
 
+		if ( is_customize_preview() ) {
+			// Scripts are always needed to inject changeset UUID.
+			$args['element_xpaths'][] = '//script';
```

Comment:
This will currently target more scripts than it needs to. It doesn't need to target JSON, for example:

```suggestion
			$args['element_xpaths'][] = '//script[ @src ]';
			$args['element_xpaths'][] = '//script[ not( @type ) or not( contains( @type, "json" ) ) ]';
```

Reply:
Also, instead of adding `element_xpaths` to the `$args` as a whole (which will pass them into _every_ sanitizer), I suggest instead putting the code in `\AMP_Theme_Support::add_hooks()` like so:

```php
if ( is_customize_preview() ) {
	add_filter(
		'amp_dev_mode_element_xpaths',
		function ( $xpaths ) {
			$xpaths[] = '//script[ @src ]';
			$xpaths[] = '//script[ not( @type ) or not( contains( @type, "json" ) ) ]';
			return $xpaths;
		}
	);
}
```

Reply:
Alternatively, it could be put into `amp_get_content_sanitizers()`:

```diff
diff --git a/includes/amp-helper-functions.php b/includes/amp-helper-functions.php
index 7166267b2..2da9428e3 100644
--- a/includes/amp-helper-functions.php
+++ b/includes/amp-helper-functions.php
@@ -1465,6 +1465,10 @@ function amp_get_content_sanitizers( $post = null ) {
 			$dev_mode_xpaths[] = '//*[ @id = "wpadminbar" ]//*';
 			$dev_mode_xpaths[] = '//style[ @id = "admin-bar-inline-css" ]';
 		}
+		if ( is_customize_preview() ) {
+			$dev_mode_xpaths[] = '//script[ @src ]';
+			$dev_mode_xpaths[] = '//script[ not( @type ) or not( contains( @type, "json" ) ) ]';
+		}
 		$sanitizers = array_merge(
 			[
 				'AMP_Dev_Mode_Sanitizer' => [
```

Reply:
Addressed in https://github.com/ampproject/amp-wp/pull/4977/commits/927ed5fd0102233fb2343993b8aa9e33ad970305.